src/Analyzers/CSharp/Analyzers/UsePrimaryConstructor/CSharpUsePrimaryConstructorDiagnosticAnalyzer.cs

```diff
@@ -381,12 +379,17 @@ void RegisterFieldOrPropertyAnalysisIfNecessary(Analyzer? analyzer)
                     Dictionary<ISymbol, IParameterSymbol> candidateMembersToRemove)
                 {
                     var semanticModel = compilation.GetSemanticModel(primaryConstructorDeclaration.SyntaxTree);
+
+                    var body = primaryConstructorDeclaration.ExpressionBody ?? (SyntaxNode?)primaryConstructorDeclaration.Body;
+                    if (body?.ContainsDirectives is true)
```

Comment:
Will this also turn off the diagnostic in presence of pragma directives? Is that desirable?

Reply:
probably.  but we can see if that's a big enough negative if people have an issue with it.